lib/mandrill.rb

```diff
@@ -29,13 +29,19 @@ def initialize(apikey=nil, debug=false)
             @apikey = apikey
         end
 
+        MAX_RETRY = 3
+
         def call(url, params={})
-            params[:key] = @apikey
-            params = JSON.generate(params)
-            r = @session.post(:path => "#{@path}#{url}.json", :headers => {'Content-Type' => 'application/json'}, :body => params)
-            
-            cast_error(r.body) if r.status != 200
-            return JSON.parse(r.body)
+          retry_count ||= 0
+          params[:key] = @apikey
+          params = JSON.generate(params)
+          r = @session.post(:path => "#{@path}#{url}.json", :headers => {'Content-Type' => 'application/json'}, :body => params)
+          JSON.parse(r.body)
+        rescue
+          raise cast_error(r.body) if (r.status != 200 && retry_count >= MAX_RETRY)
```

Comment:
no need to `raise` here as `cast_error` raises itself. also, i don't think we need the `r.status != 200` check since that should already be the case if we're in the `rescue`

Reply:
that all makes sense